Content.Shared/Projectiles/ProjectileComponent.cs

```diff
@@ -26,6 +26,12 @@ public sealed partial class ProjectileComponent : Component
     [DataField, AutoNetworkedField]
     public EntityUid? Weapon;
 
+    /// <summary>
+    ///     Ammo used to shoot this projectile.
+    /// </summary>
+    [DataField, AutoNetworkedField]
+    public EntityUid? AmmoSource;
```

Comment:
This field shouldn't be necessary from looking at the rest of the code, if you need more information you should be able to put it on the projectile with new components as well
That gets rid of most of the ammo code changes